internal/fs/local_file_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A collection of tests for a file system where we do not attempt to write to
+// the file system at all. Rather we set up contents in a GCS bucket out of
+// band, wait for them to be available, and then read them via the file system.
+
+package fs_test
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	. "github.com/jacobsa/ogletest"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type LocalFileTest struct {
+	fsTest
+}
+
+func init() {
+	RegisterTestSuite(&LocalFileTest{})
+}
+
+func (t *LocalFileTest) SetUpTestSuite() {
+	t.serverCfg.ImplicitDirectories = true
+	t.serverCfg.MountConfig = &config.MountConfig{
+		WriteConfig: config.WriteConfig{CreateEmptyFile: false},
+	}
+	t.fsTest.SetUpTestSuite()
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *LocalFileTest) TestReadDirWithEmptyLocalFile() {
+	// Create local files.
+	f1, err := os.Create(path.Join(mntDir, "localFile1"))
+	AssertEq(nil, err)
+	f2, err := os.Create(path.Join(mntDir, "localFile2"))
```

Comment:
Please validate that file isnot created in GCS and created when file.close is called.

reference: https://github.com/GoogleCloudPlatform/gcsfuse/pull/1279/files